src/main/java/de/tum/in/www1/artemis/service/tutorialgroups/TutorialGroupService.java

```diff
@@ -97,6 +97,40 @@ public void setTransientPropertiesForUser(User user, TutorialGroup tutorialGroup
             tutorialGroup.setTeachingAssistantName(null);
         }
         this.setNextSession(tutorialGroup);
+        this.setAverageAttendance(tutorialGroup);
+    }
+
+    /**
+     * Sets the averageAttendance transient field of the given tutorial group
+     * <p>
+     * Calculation:
+     * <ul>
+     * <li>Get set of the last three completed sessions (or less than three if not more available)</li>
+     * <li>Remove sessions without attendance data (null) from the set</li>
+     * <li>If set is empty, set attendance average of tutorial group to null (meaning could not be determined)</li>
+     * <li>If set is non empty, set the attendance average of the tutorial group to the arithmetic mean (rounded to integer)</li>
+     * </ul>
+     *
+     * @param tutorialGroup the tutorial group to set the averageAttendance for
+     */
+    private void setAverageAttendance(TutorialGroup tutorialGroup) {
+        Collection<TutorialGroupSession> sessions;
+        if (getPersistenceUtil().isLoaded(tutorialGroup, "tutorialGroupSessions") && tutorialGroup.getTutorialGroupSessions() != null) {
+            sessions = tutorialGroup.getTutorialGroupSessions();
+        }
+        else {
+            sessions = tutorialGroupSessionRepository.findAllByTutorialGroupId(tutorialGroup.getId());
+        }
+        sessions.stream()
+                .filter(tutorialGroupSession -> TutorialGroupSessionStatus.ACTIVE.equals(tutorialGroupSession.getStatus())
+                        && tutorialGroupSession.getEnd().isBefore(ZonedDateTime.now()))
+                .sorted(Comparator.comparing(TutorialGroupSession::getStart).reversed()).limit(3)
```

Comment:
(optional) I think the limit step can easily be overlooked as all the other steps follow a vertical flow
```suggestion
                .sorted(Comparator.comparing(TutorialGroupSession::getStart).reversed())
                .limit(3)
```